src/js/index/todolist.js

```diff
@@ -0,0 +1,121 @@
+const $todoInput = document.querySelector("#new-todo-title");
+const $todoList = document.querySelector("#todo-list");
+const $todoCount = document.querySelector("strong");
+const $filters = document.querySelector(".filters");
+
+$todoInput.addEventListener("keyup", onAddTodoItem);
+$todoList.addEventListener("click", onToggleTodoItem);
+$todoList.addEventListener("dblclick", onEditTodoItem);
+$filters.addEventListener("click", onFilter);
+
+function onAddTodoItem(event) {
+    const todoTitle = event.target.value;
+    const todoList = document.getElementById("todo-list");
+    if (event.key === "Enter" && todoTitle !== "") {
+        todoList.insertAdjacentHTML("beforeend", renderTodoItemTemplate(todoTitle));
+        updateTodoCount();
+        event.target.value = "";
+        changeState();
+    }
+}
+
+function onToggleTodoItem(event) {
+    if (event.target.classList.contains("toggle")) {
+        event.target.closest("li").classList.toggle("completed");
+        event.target.closest("input").toggleAttribute("checked");
+        changeState()
+    } else if (event.target.classList.contains("destroy")) {
+        event.target.closest("li").remove();
+        updateTodoCount();
+        changeState();
+    }
+}
+
+function onEditTodoItem(event) {
+    if (event.target.classList.contains("label")) {
+        const originalValue = event.target.innerText;
+        event.target.closest("li").classList.add("editing");
+        event.target.parentElement.parentElement.querySelector(".edit").focus();
+        event.target.closest("li")
+            .addEventListener("keyup", function (event) {editTodoItem(event, originalValue)});
+    }
+}
+
+function editTodoItem(event, originalValue) {
+    switch (event.key) {
+        case "Enter" :
+            return updateTodoItem(event)
+        case "Escape" :
+            event.target.value = originalValue;
+            return event.target.closest("li").classList.remove("editing");
+    }
+}
+
+function updateTodoItem(event) {
+    const todos = document.querySelectorAll("#todo-list > li");
```

Comment:
```suggestion
    const $$todos = document.querySelectorAll("#todo-list > li");
```

HTMLElement 를 가르키는 변수에 $ prefix 를 추가하는 컨벤션을 적용하셨으면, 일관되게 적용하시는게 좋습니다!
querySelectorAll 을 사용하여 셀렉트한 HTMLElement 배열에 대하여는 $$ prefix 를 붙이는게 일반적입니다! 
($를 붙이는 사람도 있을것 같긴 합니다만,, 저는 $$ 가 좋은것 같아요 ㅎㅎ)